lib/std/system.nim

```diff
@@ -75,6 +75,11 @@ proc low*[I, T](x: typedesc[array[I, T]]): I {.magic: "Low", noSideEffect.}
 proc high*[T: Ordinal|enum|range](x: typedesc[T]): T {.magic: "High", noSideEffect.}
 proc high*[I, T](x: typedesc[array[I, T]]): I {.magic: "High", noSideEffect.}
 
+proc `[]`*[I, T](x: array[I, T], i: I): T {.magic: "ArrAt".}
+proc `[]`*(x: string, i: int): char {.magic: "StrAt".}
```

Comment:
That is just some `.inline` proc, we need to be able to acces the `string` as its `(ptr, len)` pair though inside the proc body.

Reply:
Excluded string for now